### PR TITLE
Informs if upload was Multipart

### LINKED ISSUE #1932: ManagedUpload - Flag when file was uploaded using multipart
Currently we rename the files based on the ETag after upload and then do a Check if the Filename is the same as ETag, but the returned ETag in the ManagedUpload is the "Special Multipart" one.
Is it possible to make the result in `data` var include sometype of flag like `UploadedViaMultipart=true` so we can calculate the real ETag ?

Thanks

### PR DESCRIPTION
**Pull request by driverpt**
Fixes #1932

### CODE REVIEW

**Comment by jeskew** [comment]
We generally prefix injected parameters with a dollar sign to avoid overwriting any future parameters added to the service response.

Is the intention here to be able to use the ETag value as a checksum?

**Comment by driverpt** [comment]
The purpose is the client uploads directly to S3 and use it as a Transient Store. After the client uploads it, we rename the file to it's checksum, in this case, when we're copying/renaming the object it gets a new checksum. I think it would be best if the SDK indicates in the callback if the file was uploaded via Multipart.

File: lib/s3/managed_upload.js [hunk 1]
```diff
@@ -634,6 +634,10 @@ AWS.S3.ManagedUpload = AWS.util.inherit({
         data.Location = data.Location.replace(/%2F/g, '/');
       }
 
+      if (data) {
+        data.UploadedViaMultipart = true;
```
Comment (AllanFly120) [inline lib/s3/managed_upload.js]:
```suggestion
        data.$UploadedViaMultipart = true;
```

**Comment by AllanFly120** [comment]
Hey @driverpt Are you still interested in working on this PR?

**Comment by AllanFly120** [comment]
@driverpt 
Thanks. I prefer the lowercased flag: `$uploadedViaMultipart` instead of `$UploadedViaMultipart`. It's a little nit-picky but this will make the casing consistent. 

Can you also add change log for this feature? Follow [this checklist](https://github.com/aws/aws-sdk-js/blob/master/.github/pull_request_template.md) to see what else is needed. Personally I think updating [corresponding](https://github.com/aws/aws-sdk-js/blob/master/lib/s3/managed_upload.d.ts#L43) typescript interface would be very helpful.

**Pull request closed without merging**